File: package-testing/php-sdk-relay/src/RelayLogger.php
<?php

namespace Eppo\SDKTest;

use Eppo\Logger\AssignmentEvent;
use Eppo\Logger\BanditActionEvent;
use Eppo\Logger\IBanditLogger;

class RelayLogger implements IBanditLogger
{
    /**
     * @var AssignmentEvent[]
     */
    public array $assignmentLogs = [];

    /**
     * @var BanditActionEvent[]
     */
    public array $banditLogs = [];

    public function logAssignment(AssignmentEvent $assignmentEvent): void
    {
        $this->assignmentLogs[] = $assignmentEvent;
    }

    public function logBanditAction(BanditActionEvent $banditActionEvent): void
    {
        $this->banditLogs[] = $banditActionEvent;
    }

    public function resetLogs(): void
    {
        $this->assignmentLogs = [];
        $this->banditLogs = [];
    }
}

File: package-testing/dart-sdk-relay/lib/relay_logger.dart
import 'package:logging/logging.dart';

// This class will need to be updated based on the actual Eppo Dart SDK implementation
class RelayLogger {
  final _logger = Logger('RelayLogger');
  
  // Store assignment logs
  final List<Map<String, dynamic>> assignmentLogs = [];
  
  // Store bandit logs
  final List<Map<String, dynamic>> banditLogs = [];
  
  // Log an assignment event
  void logAssignment(Map<String, dynamic> assignmentEvent) {
    _logger.fine('Logging assignment: $assignmentEvent');
    assignmentLogs.add(assignmentEvent);
  }
  
  // Log a bandit action event
  void logBanditAction(Map<String, dynamic> banditActionEvent) {
    _logger.fine('Logging bandit action: $banditActionEvent');
    banditLogs.add(banditActionEvent);
  }
  
  // Reset all logs
  void resetLogs() {
    assignmentLogs.clear();
    banditLogs.clear();
  }
} 